torch/_inductor/scheduler.py

```diff
@@ -1405,7 +1449,7 @@ def add_user(used_by_name, user_node, can_inplace=False, is_weak=False):
                 # this node must run after the prior writer
                 add_user(alt_name, node)
                 node.add_mutation_dep(StarDep(alt_name))
-                for other_node in name_to_users[alt_name]:
+                for other_node in name_to_users[alt_name].items:
```

Comment:
Can you explain why appending in the middle of an iteration is required here? From what I understand, we are iterating over the users of node A, and appending to their respective lists, and because node A is not dependent on itself, we should never be touching the list we are iterating over, so I would think a set should work here.

Reply:
I tried using a dict first (since sets do not have deterministic iteration order) and got the error message "dict was modified during iteration". So it looks like `add_user` below somehow modifies name_to_users for the node we are iterating over

Reply:
Which test surfaced this error? Just curious

Reply:
It is a custom kernel on D51815383